File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.*;
class Solution {
    public List<Integer> solution(String[] name, int[] yearning, String[][] photo) {
        List<Integer> ret = new ArrayList<>();
        
        for(int i = 0; i < photo.length; i++){
            int sum = 0;
            for(int j = 0; j < photo[i].length; j++){
                for(int k = 0; k < name.length; k++){
                    if(photo[i][j].equals(name[k])) sum += yearning[k];
                }
            }
            ret.add(sum);
        }
        
        return ret;
    }
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    temp = {x1:x2  for x1, x2 in zip(name, yearning)}
    ret = []
    for p in photo:
        ret. append(sum(temp[nn] for nn in p if temp.get(nn)))
    return ret
    
    
"""
추억점수 = 그리움 점수의 총합 

"""